ocs/volumes.py

```diff
@@ -0,0 +1,252 @@
+"""
+A module which consists of kubevolume related operations
+"""
+
+import os
+import logging
+
+from ocs import defaults
+from ocs import kinds
+from kubernetes import config
+from openshift.dynamic import DynamicClient
+from utility.utils import get_random_str
+from ocs.rook import RookCluster
+from utility.templating import generate_yaml_from_jinja2_template_with_data
+
+logger = logging.getLogger(__name__)
+
+THIS_DIR = os.path.dirname(os.path.abspath(__file__))
+TOP_DIR = os.path.dirname(THIS_DIR)
+TEMPLATE_DIR = os.path.join(TOP_DIR, "templates/ocs-deployment/")
+
+
+class KubeVolume(object):
+    """
+    Base class for cluster volumes
+
+    Attributes:
+        name (str): name of the RBD volume
+        namepsace (str): namespace to create RBD volume
+
+    """
+    def __init__(self, name, namespace):
+        k8s_client = config.new_client_from_config()
+        self.dyn_client = DynamicClient(k8s_client)
+        self.name = name
+        self.namespace = namespace
+
+
+class CephRBDVolume(KubeVolume):
+    """
+    Class which contains Ceph RBD related functionality
+
+    Attributes:
+        name (str): name of the RBD volume
+        namepsace (str): namespace to create RBD volume
+
+    """
+    def __init__(self, name=None, namespace='default'):
+        super().__init__(name, namespace)
+        self.kind = kinds.CEPHBLOCKPOOL
+        self.api_version = defaults.OPENSHIFT_REST_CLIENT_API_VERSION
+        self.service_cbp = self.dyn_client.resources.get(
+            api_version=self.api_version,
+            kind=self.kind
+        )
+        self.template_path = os.path.join(TEMPLATE_DIR, "cephblockpool.yaml")
+        self.rk = RookCluster()
+
+    def create_cephblockpool(
+        self,
+        cephblockpool_name_prefix="autotests-cephblockpool",
+        failureDomain="host",
+        size="3"
+    ):
+        """
+        Creates cephblock pool
+
+        Args:
+            cephblockpool_name_prefix (str): prefix given to cephblockpool
+            failureDomain (str): The failure domain across which the
+                                   replicas or chunks of data will be spread
+            size (int): The number of copies of the data in the pool.
```

Comment:
The default you passed is str - "3"
And because this is not the volume size, maybe change it to replica_count

Reply:
make sense to change to replica_count

Reply:
Done